tests/v2/test_utils_keyfile.nim

```diff
@@ -0,0 +1,379 @@
+{.used.}
+
+import
+  std/[json, os],
+  testutils/unittests, chronos,
+  eth/keys
+import
+  ../../waku/v2/utils/keyfile
+
+from ../../waku/v2/protocol/waku_noise/noise_utils import randomSeqByte
+
+suite "KeyFile test suite":
+
+  let rng = newRng()
+
+  test "Create/Save/Load single keyfile":
+
+    # The password we use to encrypt our secret
+    let password = "randompassword"
+
+    # The filepath were the keyfile will be stored
+    let filepath = "./test.keyfile"
+    defer: removeFile(filepath)
+
+    # The secret
+    var secret = randomSeqByte(rng[], 300)
+
+    # We create a keyfile encrypting the secret with password
+    let keyfile = createKeyFileJson(secret, password)
+
+    check:
+      keyfile.isOk()
+      # We save to disk the keyfile
+      saveKeyFile(filepath, keyfile.get()).isOk()
+
+    # We load from the file all the decrypted keyfiles encrypted under password
+    var decodedKeyfiles = loadKeyFiles(filepath, password)
+
+    check:
+      decodedKeyfiles.isOk()
+      # Since only one secret was stored in file, we expect only one keyfile being decrypted
+      decodedKeyfiles.get().len == 1
+
+    # We check if the decrypted secret is the same as the original secret
+    let decodedSecret = decodedKeyfiles.get()[0]
+
+    check:
+      secret == decodedSecret.get()
+
+  test "Create/Save/Load multiple keyfiles in same file":
+
+    # We set different passwords for different keyfiles that will be stored in same file
+    let password1 = "password1"
+    let password2 = ""
+    let password3 = "password3"
+    var keyfile: KfResult[JsonNode] 
+
+    let filepath = "./test.keyfile"
+    defer: removeFile(filepath)
+
+    # We generate 6 different secrets and we encrypt them using 3 different passwords, and we store the obtained keystore
+
+    let secret1 = randomSeqByte(rng[], 300)
```

Comment:
Since there is a clear pattern here, perhaps just use a for loop?

```nim
for pass in @[pass1, passn]:
  let secret = randomSeqByte(rng[], 300)
  let keyfile = createKeyFileJson(secret, pass)
  check:
      keyfile.isOk()
      saveKeyFile(filepath, keyfile.get()).isOk()
```

you can even randomize the password for each iteration, or even include %,Aa-? and weird stuff just in case?